### DISCUSSION TITLE
Thoughts on making username matching case-insensitive

### DISCUSSION THREAD

**Discussion by DavidBrunow** [danger/danger-js#1289] [Ideas]
Yesterday I spent about half my day tracking down an issue where our BitBucket Server was not deleting previous inline comments and therefore each time our build jobs ran it would add the same comments that had already been added. The root cause was that the username we were using to run Danger, for example “sampleusername”, was not capitalized the same way as the username configured in BitBucket Server, for example “sampleuserNAME”.

Since BitBucket Server (and I believe all other hosts although I don’t have experience with GitLab) treats usernames as case-insensitive should we change Danger’s behavior as well?

Here is the line of code that I eventually found at the end of my debugging journey yesterday:

https://github.com/danger/danger-js/blob/main/source/platforms/bitbucket_server/BitBucketServerAPI.ts#L235

Thoughts on `toLowerCase()`ing both the usernames there?

**Comment by orta** [comment]
Yep, makes sense to me 👍🏻

**Comment by DavidBrunow** [comment]
Great I will get a pull request up when I get a chance in the near future.

**Comment by DavidBrunow** [comment]
I looked at all the other providers and they compare IDs instead of usernames. My changes will be limited to BitBucket Server.
